libdnf/dnf-context.cpp

```diff
@@ -364,6 +364,7 @@ dnf_context_init(DnfContext *context)
     priv->install_root = g_strdup("/");
     priv->check_disk_space = TRUE;
     priv->check_transaction = TRUE;
+    // TODO(jkolarik): Remove when the system-wide change proposal is accepted
```

Comment:
I don't think that removal of the attribute is a good idea. There are API methods that use it - `dnf_context_set_enable_filelists`. I am proposing to set it `False` and sync the context setting with `dnf.conf`.

We may add new value (need to check whether it brakes ABI compatibility - it is glib private structure)

`priv->enable_filelists_updated_from_conf = FALSE;`
```
gboolean
dnf_context_get_enable_filelists (DnfContext     *context)
{
    DnfContextPrivate *priv = GET_PRIVATE(context);
    if (!priv->enable_filelists_updated_from_conf) {
        auto & optional_metadata_types = libdnf::getGlobalMainConfig().optional_metadata_types().getValue();
        // Test the content of configuration and update priv->enable_filelists
        priv->enable_filelists_updated_from_conf = TRUE;
    }
    return priv->enable_filelists;
}
```

Then we don't require any other changes in the code. What do you think?

Reply:
> I don't think that removal of the attribute is a good idea. 

Oh, I just wanted to remove the line `priv->enable_filelists = TRUE;` as the `enable_filelists` are `FALSE` by default if I see it correctly.

But when I am thinking about it, why don't just use the existing `dnf_context_get_enable_filelists` and `dnf_context_set_enable_filelists` to load and store the value from configuration. We can keep the private member `enable_filelists` unused with some deprecation note to not break the ABI. Or am I missing something here?

Reply:
This sound like a good idea. But I have some doubts. I am not sure whether `auto & optional_metadata_types = libdnf::getGlobalMainConfig()` could be modified by multiple instances of context. PackageKit is a daemon  and I am not sure what resources are shared. @jrohel May I ask you whether updating values of `GlobalMainConfig()` cannot interfere with other instances of context running by the same process?

Reply:
PackageKit ignores some parts of the libdnf API and does its own initialization. PackageKit does not use the `dnf_context_set_enable_filelists` function.

Instead, PackageKit sets its `flags` variable in its own way and calls the libdnf API function `dnf_sack_add_repos` with it. And I think the `DNF_SACK_ADD_FLAG_FILELISTS` flag is always set in the `flags` variable. Hardcoded in PackageKit source code in function `dnf_utils_create_sack_for_filters`.

Reply:
Thanks for the feedback. I am switching to the simple config load/store approach and I'll create a related upstream issue for PackageKit.